studio/schemas/objects/link.ts

```diff
@@ -190,10 +192,12 @@ export const link = defineField({
       title: "linkTitle",
       type: "linkType",
     },
-    prepare(selection) {
-      const { title, type } = selection;
+    prepare({ title, type }) {
+      const translatedTitle = isInternationalizedString(title)
```

Comment:
Because we know that link is of type  `internationalizedArrayString`, we don't need this check.

Reply:
We don't actually know that though. More importantly, TypeScript doesn't know what type it is (since `prepare` gives us variables of type `any`). With the type guard, we can safely ignore the value if it doesn't meet our expectations, and otherwise use the variable with proper typing (which makes the use of `firstTranslation` much nicer). Without the type guard, we are using array operations on something that might not be an array and accessing properties that might not exist.

This is of course a question of how "safe" we want to be, but since we have favored the safer side [before](https://github.com/varianter/variant.no/pull/535#discussion_r1744997509) I think this check fits nicely.

Reply:
Why don't we know it's of type internationalizedArrayString when that's why we're setting on the schema? 

I agree however that there should be a better approach given cases where we update the schema type but forget to adjust in preview. Could we leave it as it was and come up with a better approach for all preview cases? Because this wouldn't just concern string array, but cases where we say adjust the schema name that's used as reference in a preview

Reply:
> Why don't we know it's of type internationalizedArrayString when that's why we're setting on the schema?

Because there is no typing of the preview variables. So the schema could be changed without raising any errors. With the type guard we can try to give the variable a proper type, and safely fail if the schema has changed. Using a variable with an actual type is much easier than using an `any` variable, which could be anything. We could also use the `as` keyword to achieve this, but then we would just bypass the type check completely and possibly end up with undefined behavior. 

> I agree however that there should be a better approach given cases where we update the schema type but forget to adjust in preview. Could we leave it as it was and come up with a better approach for all preview cases? Because this wouldn't just concern string array, but cases where we say adjust the schema name that's used as reference in a preview

This is an attempt at checking that. I agree that this check would be nice in other uses of `prepare`, but I don't see why that should stop us from doing it only here as a start.

Reply:
My concern is that this approach is too specific to this scenario and that it wouldn't transfer nicely in other scenarios, giving us multiple type guards. The main problem is that our checks aren't triggered if preview get's an error. How about we start there first before we add type guards?

Reply:
I don't see why we shouldn't use a type guard just because it doesn't solve all our problems related to preview. What is your concern with multiple type guards? A type guard is only for one type, so naturally you would need multiple type guards to handle different types.

Are you maybe thinking that we should throw an error instead of failing silently? We could still do that if the type guard returns false.

Reply:
My concern is that the handling of previews are done manually right now. There are no checks or tests that fail if preview throws an error such as `Invalid Preview Config`. So if a dev ads changes to the schema, and specifically changes schema values that are used in preview, we have to rely on our memory to update after changes.

So a type guard is nice and all as it gives us visually a fallback, but we're still not alerted to the changes done which preview has to account for

Reply:
Added a TypeError that throws if the type guard fails. It will then display an error in the preview component.

<img width="615" alt="image" src="https://github.com/user-attachments/assets/8b0f91d4-05bf-4128-bf9d-add7e8e2d63e">